enhancements/storage/vsphere-storage-check.md

```diff
@@ -0,0 +1,232 @@
+---
+title: vsphere-monitoring-operator
+authors:
+  - "@jsafrane"
+reviewers:
+  - "@hekumar"
+approvers:
+  - TBD
+creation-date: 2020-11-10
+last-updated: 2020-11-10
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# vSphere monitoring for OCP clusters
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Below we propose to check if OpenShift clusters installed into vSphere environment is / can be fully functional by introducing a new operator and periodical checks.
+
+## Motivation
+
+Our support frequently faces OCP clusters on vSphere that is either mis-configured.
+
+* OCP lacks necessary permissions: [#1869035](https://bugzilla.redhat.com/show_bug.cgi?id=1869035) [#1879959](https://bugzilla.redhat.com/show_bug.cgi?id=1879959). [#1852304](https://bugzilla.redhat.com/show_bug.cgi?id=1852304).
+* Datastore used in storage classes / OCP config is too long: [#1884202](https://bugzilla.redhat.com/show_bug.cgi?id=1884202).
+* Users don't enable `disk.enableUUID=TRUE`: [#1434709](https://bugzilla.redhat.com/show_bug.cgi?id=1434709) [1850984](https://bugzilla.redhat.com/show_bug.cgi?id=1850984) [1727991](https://bugzilla.redhat.com/show_bug.cgi?id=1727991).
+* Kubelet is configured without `--cloud-provider=vsphere`: [1632729](https://bugzilla.redhat.com/show_bug.cgi?id=1632729).
+* Storage Distributed Resource Scheduler (DRS) is not supported.
+* vMotion (i.e. VM migration) is not supported.
```

Comment:
@gnufied with your recent converstation with vmware folks I thought vmotion wasn't an issue. We know that **storage** vmotion is not supported

Reply:
Yep, it might be trickier to detect something like that in a monitoring tool though...

Reply:
If storage drs is enable we could know that. If a storage vmotion we would just have tasks and events to monitor.

Reply:
For a VM migration, I don't see an event generated at all. And for storage migration, our VMC does not allow that (there's only one datastore)

Reply:
@jsafrane There is one upstream metric which was merged yesterday, which will allow tracking control-plane node (only) migrations between vCenters in a multi-vcenter cluster configuration. Just did a cherry pick on that - https://github.com/openshift/kubernetes/pull/447 But the main goal was to show vCenter version, this is just a side-effect... Is this something that this proposal might need?

Reply:
Yes, vCenter version should be reported - I added it to the list of metrics.